scripts/deploy_ethers.js

```diff
@@ -1,29 +1,43 @@
 // Right click on the script name and hit "Run" to execute
-(async () => {
-    try {
-        console.log('Running deployWithEthers script...')
-    
-        const contractName = 'Storage' // Change this for other contract
-        const constructorArgs = []    // Put constructor args (if any) here for your contract
-
-        // Note that the script needs the ABI which is generated from the compilation artifact.
-        // Make sure contract is compiled and artifacts are generated
-        const artifactsPath = `browser/contracts/artifacts/${contractName}.json` // Change this for different path
-    
-        const metadata = JSON.parse(await remix.call('fileManager', 'getFile', artifactsPath))
-        // 'web3Provider' is a remix global variable object
-        const signer = (new ethers.providers.Web3Provider(web3Provider)).getSigner()
-    
-        let factory = new ethers.ContractFactory(metadata.abi, metadata.data.bytecode.object, signer);
-    
-        let contract = await factory.deploy(...constructorArgs);
-    
-        console.log('Contract Address: ', contract.address);
-    
-        // The contract is NOT deployed yet; we must wait until it is mined
-        await contract.deployed()
-        console.log('Deployment successful.')
-    } catch (e) {
-        console.log(e.message)
-    }
-})()
\ No newline at end of file
+const hre = require("hardhat")
+
+const main = async () => {
+  console.log('Running deployWithEthers script...')
+  /*
+  const ERC20OA = await hre.ethers.getContractFactory('ERC20OA')
+  const erc20OA = await ERC20OA.deploy(
+    'test', // Name token
+    'tst', // Symbol token
+    '0xAE8C6d624bC59F1B7CeA1F6d17255A995930a31F', // Address Admin
+    '0x794Fed8D9bEe97a7e352e4C308Cec5632b741104', //Address Marketing
+    '0x8A7e80fDF8c8c6f236d6e230F99633869AB8C4de', // Address Development
+    '0xbef2576102146b7BcAb08DD39c195aeD21372998', // Helping address,
+    '0x1F34D46f8467429a7a4fdFb37D0Cb825f3aeD48F', // Rewards Address
+  )
+
+  await erc20OA.deployed()
+
+  const erc20OAAddress = erc20OA.address
+  console.log(`ERC20OA smart contract deployed at ${erc20OAAddress}`)
+  */
+
+  const PresaleOA = await hre.ethers.getContractFactory('PreSale')
+  const presale = await PresaleOA.deploy(
+    '0xD88e24aff1598ac452c7B2C4CA56db653053cd62', // Address of token to hold
+    '1660923000' // End time in epoch time
+  )
+
+  await presale.deployed()
+
+  const presaleAddress = presale.address
+  console.log(`Presale smart contract deployed at ${presaleAddress}`)
+
+  console.log('Deployment successful.')
+}
+
+main ()
```

Comment:
I have a question, is necesary return a promise in function syncro? It 'cause you are waiting a promise, when you code on main() is syncro, and the function wait a callback async. If you want to catch a error o success response, you can use try and catch. 

https://stackoverflow.com/questions/35302431/async-await-implicitly-returns-promise

Reply:
I use a promise because main function is asynchronous because it is waiting until one smart contract is deployed to start the deploy of the second (This because the second  contract needs the address of the first one to be deployed).